extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -116,7 +121,8 @@ public OpenCensusProtobufInputRowParser withParseSpec(ParseSpec parseSpec)
     for (TimeSeries ts : metric.getTimeseriesList()) {
 
       // Add common resourceLabels.
-      Map<String, Object> labels = new HashMap<>(metric.getResource().getLabelsMap());
+      Map<String, Object> labels = new HashMap<>(metric.getResource().getLabelsMap().entrySet().stream()
+          .collect(Collectors.toMap(entry -> resourceLabelsKeyMapper(entry.getKey()), Map.Entry::getValue)));
```

Comment:
it seems unnecessary to prefix the labels a second time when we've already done so when updating `recordDimensions`.

Reply:
A complete miss by me, thanks for pointing out.

Reply:
Optimized further with single key mapper for each type, processed once and used either the `key sets` or `label keys list`.